### PR TITLE
Directive validation

### PR DESCRIPTION
**Pull request by shrverma**

### CODE REVIEW

File: wrangler-api/src/main/java/io/cdap/wrangler/api/DirectiveRelationalTransform.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright © 2017-2019 Cask Data, Inc.
```
Comment (vanathi-g) [inline wrangler-api/src/main/java/io/cdap/wrangler/api/DirectiveRelationalTransform.java]: nit: fix license file

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 2]
```diff
@@ -524,6 +531,34 @@ private void validatePrecondition(String precondition, Boolean isConditionSQL, F
...
 
   private void validateSQLModeDirectives(FailureCollector collector) {
+
+    String recipe = config.getDirectives();
+
+    registry = SystemDirectiveRegistry.INSTANCE;
+    try {
+      registry.reload("default");
+    } catch (DirectiveLoadException e) {
+      throw new RuntimeException(e);
```
Comment (vanathi-g) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java]: Add more descriptive messages to all the errors if re-throwing or alternatively just add to failure collector. Can also merge the catch blocks if the errors are handled the same way: Ex: catch (DirectiveParseException | RecipeException e)

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 3]
```diff
@@ -50,6 +50,7 @@
 import io.cdap.wrangler.api.Directive;
 import io.cdap.wrangler.api.DirectiveLoadException;
 import io.cdap.wrangler.api.DirectiveParseException;
+import io.cdap.wrangler.api.DirectiveRelationalTransform;
```
Comment (vanathi-g) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java:52]: nit: remove unused imports

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 4]
```diff
@@ -524,6 +531,34 @@ private void validatePrecondition(String precondition, Boolean isConditionSQL, F
   }
 
   private void validateSQLModeDirectives(FailureCollector collector) {
```
Comment (vanathi-g) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java]: Can change the function to take a list of directives and check whether they are sql supported. It seems like the function performs too many different operations

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 5]
```diff
@@ -524,6 +531,34 @@ private void validatePrecondition(String precondition, Boolean isConditionSQL, F
...
+
+    String recipe = config.getDirectives();
+
+    registry = SystemDirectiveRegistry.INSTANCE;
```
Comment (vanathi-g) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java]: Also, this logic of loading directives is repeated in multiple places in this file. Make it a separate function?

**Review by vanathi-g — CHANGES_REQUESTED** [review]

**Review by vanathi-g — APPROVED** [review]

**Review by saimukkamala — APPROVED** [review]

**Pull request merged**